src/main/scala/metadata.scala

```diff
@@ -0,0 +1,83 @@
+package suffuse
+
+/** AKey is an attribute key: a Metadata map key which carries
+ *  both the type of the corresponding value, and a value to
+ *  provide when there is no value in the map. This as opposed
+ *  to wrapping everything in sight in Option.
+ */
+final class AKey[A](description: String, empty: => A) {
+  def emptyValue: A = empty
```

Comment:
> `K => V` and not `K => Option[V]`, with total functions instead of partial functions

They are both total functions as in "defined for all inputs of the right type", I get what you mean though. I however do not have the experience where that stands in the way that can not not be solved with some helper utilities.

Note that I am willing to go your route, I am however skeptical.